src/registers/model_specific.rs

```diff
@@ -636,4 +702,44 @@ mod x86_64 {
             Self::write(flags, legacy_bitmap);
         }
     }
+
+    impl Pat {
+        /// Reads IA32_PAT.
+        ///
+        /// # Safety
+        ///
+        /// The PAT must be supported on the CPU, otherwise a general protection exception will
+        /// occur. Support can be detected using the `cpuid` instruction.
+        #[inline]
+        pub unsafe fn read() -> [PatFlags; 8] {
```

Comment:
Let's make the read function safe. PAT support is broadly available and we don't consider #GP exceptions to be UB. Most other MSR wrappers also have safe read functions.